iceoryx2-cli/iox2/src/cli.rs

```diff
@@ -0,0 +1,30 @@
+use clap::Parser;
+
+#[derive(Parser, Debug)]
+#[command(
+    name = "iox2",
+    about = "The command-line interface to iceoryx2",
+    long_about = None,
+    version = env!("CARGO_PKG_VERSION"),
+    disable_help_subcommand = true,
+    arg_required_else_help = true,
+    help_template = help_template(),
+)]
+pub struct Cli {
+    #[arg(short, long, help = "List all installed commands")]
+    pub list: bool,
+
+    #[arg(short, long, help = "Run development commands")]
+    pub dev: bool,
+
+    #[arg(hide = true, required = false)]
+    pub external_command: Vec<String>,
+}
+
+fn help_template() -> &'static str {
```

Comment:
@orecham I thought clap is generating the whole help automatically. Was it not compatible with commands?

Reply:
@elfenpiff The problem is that clap cannot dynamically generate the help, which is required if the available commands are to be determined by the binaries that are present.